src/common/Bdoc10Handler.cpp

```diff
@@ -1,76 +1,39 @@
 #include "Bdoc10Handler.h"
 
-#include <QXmlStreamReader>
 #include <zip.h>
 
-//Verifies that the file provided by parameter filePath is a BDOC10 conatiner.
+//Verifies that the file provided by parameter filePath is a BDOC10 container.
 bool Bdoc10Handler::isBdoc10Container(QString const& filePath) {
-    const QString BDOC_MANIFEST_PATH = "META-INF/manifest.xml";
+    const char* BDOC_MIMETYPE_PATH = "mimetype";
+    const char* MIMETYPE_BDOC10_APPLICATION = "application/vnd.bdoc-1.0";
 
     bool bRes = false;
     int errorp = 0;
     zip *zipConatiner = zip_open(filePath.toUtf8().constData(), ZIP_RDONLY, &errorp);
     if (NULL != zipConatiner) {
-	    zip_int64_t index = zip_name_locate(zipConatiner, BDOC_MANIFEST_PATH.toUtf8().constData(), 0);
+        zip_int64_t index = zip_name_locate(zipConatiner, BDOC_MIMETYPE_PATH, 0);
 	    if (-1 != index) {
 		    zip_file_t *zipfile = zip_fopen_index(zipConatiner, index, ZIP_FL_UNCHANGED);
 		    if (NULL != zipConatiner) {
-			    const int FILE_CHUNK_SIZE = 1024;
+			    const int FILE_CHUNK_SIZE = 256;
 			    char buf[FILE_CHUNK_SIZE];
-			    QByteArray xmlData;
+			    QByteArray data;
 			    for (;;)
 			    {
 				    zip_int64_t read = zip_fread(zipfile, buf, FILE_CHUNK_SIZE);
```

Comment:
Why not read only same amount as sizeof(MIMETYPE_BDOC10_APPLICATION).
No need read full file
```
QByteArray data(sizeof(MIMETYPE_BDOC10_APPLICATION), 0);
zip_int64_t read = zip_fread(zipfile, &data[0], data.size());
if(data == MIMETYPE_BDOC10_APPLICATION)
```